src/main/scala/firrtl/EmissionOption.scala

```diff
@@ -0,0 +1,63 @@
+// See LICENSE for license details.
+
+package firrtl
+
+/** 
+  * Base type for emission customization options 
+  * NOTE: all the following traits must be mixed with SingleTargetAnnotation[T <: Named]
+  *       in order to be taken into account in the Emitter
+  */
+trait EmissionOption
+
+
+/** Emission customization options for registers */
+trait RegisterEmissionOption extends EmissionOption {
+  /** when true the reset init value will be used to emit a bitstream preset */
+  def useInitAsPreset : Boolean = false
+  
+  /** when true the initial randomization is disabled for this register */
+  def disableRandomization : Boolean = false
+}
+
+/** default Emitter behavior for registers */
+case object RegisterEmissionOptionDefault extends RegisterEmissionOption 
+
+
+/** Emission customization options for IO ports */
+trait PortEmissionOption extends EmissionOption {
+  /** force removal of the current port */
+  def forceRemoval : Boolean = false 
```

Comment:
Per conversation in dev meeting on 2 March 2020, `forceRemoval` is a bit of a dangerous notion to expose to the user. Perhaps instead we should standardize some notion of `garbage` (that is more generic to other FIRRTL transforms) for nodes that are introduced by FIRRTL transforms that should be DCE-d even if optimizations are turned off.